### PR TITLE
#4236 Store alias to contacts only when prefered

### LINKED ISSUE #4236: don't save alias to contacts during setup if chosen not to sumit it to attester
Description:
The browser extension throws an error when trying to save an IDN email into the contacts. This prevents the user from creating and importing their private key even when they choose not to use the email alias at all (unchecked).

They could have ended up removing the email alias from their Gmail account but I think it was not a good/practical option for them.

Consider the following js snippet:
```javascript
function isEmailValid(email) {
    if (email.indexOf(' ') !== -1) {
      return false;
    }
    email = email.replace(/\:8001$/, '');
    return /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i.test(email);
  }
var email = "demo@åwesomedomain.com"
var email2 = "åwesome@awesomedomain.com"
console.log(isEmailValid(email)) // false
console.log(isEmailValid(email2)) // true
  ```
Screenshot:

Reference: https://mail.google.com/mail/u/1/#inbox/FMfcgzGmtFJRDCbGVwrvFQXDZXlQczDv

Thanks,
Mart Gil

### PR DESCRIPTION
**Pull request by martgil**
This PR updates `saveKeysAndPassPhrase()` and `addOrReplaceKeysAndPassPhrase()` in `helper.ts` by improving the process of updating contacts whenever the user decides not to submit public key(s) for an email alias.

close #4236

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
Hello @sosnovsky, this is now ready for review. thank you!

File: test/source/tests/setup.ts [hunk 1]
```diff
@@ -123,6 +123,11 @@ export const defineSetupTests = (testVariant: TestVariant, testWithBrowser: Test
         });
         expect(t.context.mockApi!.configProvider?.config.attester?.pubkeyLookup?.[acct]).not.to.be.an('undefined');
         expect(t.context.mockApi!.configProvider?.config.attester?.pubkeyLookup?.['flowcryptcompatibility@gmail.com']).not.to.be.an('undefined');
+        // Additional test for https://github.com/FlowCrypt/flowcrypt-browser/pull/5633
+        await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+        const contactsFrame = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, '@action-open-contacts-page', ['contacts.htm', 'placement=settings']);
+        expect(await contactsFrame.isElementPresent('@action-show-email-flowcryptcompatibilitygmailcom(flowcrypt.compatibility)')).to.be.true; // main user email
+        expect(await contactsFrame.isElementPresent('@action-show-email-flowcryptcompatibilitygmailcom')).to.be.true; // user email alias
```
[the remark this answers is not in the document]
Comment (martgil) [reply test/source/tests/setup.ts]: Hi @sosnovsky, thanks for the remarks. I missed it but now I have updated a test cases for public keys that are submitted/not submitted.

**Comment by martgil** [comment]
@sosnovsky ready for review again!

**Review by sosnovsky — APPROVED** [review]
Well done 👍

**Pull request merged**